### ISSUE TITLE
Support 2FA

### ISSUE THREAD

**Issue by mclarence** [Ombrelin/plex-rich-presence#12]
Suport two factor authentication login. Currently it partially works by entering the 2FA code at the end of the password (As outlined [here](https://support.plex.tv/articles/two-factor-authentication/) under "Old Third-Party Apps & Tools". However when the session/user info is requested it returns unauthorised.

Possibly support newer authentication method outlined [here](https://forums.plex.tv/t/authenticating-with-plex/609370)? Using this newer auth method requires a lot of code to be changed. I'm not too experienced with the way http requests are handled in this program so I cant really implement it my self (unless you can point me in the right direction).

**Comment by Ombrelin** [comment]
Yes I understand, the api calls are a bit messy now and after tackling the issue with multiple plex server this week (I hope), I plan to refactor the code using MVVM pattern and RxJava, to keep the code clean and readable as we extend the app. I will look into 2FA in the meantime, and thank you for the input. I'll keep you posted !

**Comment by Ombrelin** [comment]
I also added a wiki page with the call sequence to fetch the session : https://github.com/Ombrelin/plex-rich-presence/wiki/Technical-Documentation

**Comment by mclarence** [comment]
Thanks for that diagram.

Have you started transitioning to rxjava? 

I've tried implementing the new authentication method but I am failing miserably. I only got as far as making a http request. Having issues with parsing the response. Anyways this is to be expected as I barely know anything about retrofit. Like you said, the API calls are all over the place which is rather making my attempt at this difficult. 🤣

**Comment by Ombrelin** [comment]
Not yet, I hope to start to tackle the RxJava transition this weekend. My foremost goal was to fix the multiple server thing since it was blocking for some users.
The main problem that I aim to solve with RxJava is the callback hell induced by the JavaFX concurrency API.

**Comment by Ombrelin** [comment]
Okay so I had extra time today and was able to completly refactor the app using RxJava and MVVM Pattern. The whole code should be much easier to work with now. The new version is available on `develop`. I'll merge it with the next new feature.
